blocklib/picoscope/Picoscope.hpp

```diff
@@ -540,20 +531,20 @@ struct Picoscope : public gr::Block<TPSImpl, gr::BlockingIO<true>, gr::Supported
         triggerTags.reserve(triggerOffsets.size());
 
         for (const auto triggerOffset : triggerOffsets) {
-            triggerTags.emplace_back(static_cast<int64_t>(state.produced_worker + triggerOffset), gr::property_map{ // TODO use data from timing message
-                                                                                                                    { gr::tag::TRIGGER_NAME, "PPS" },
-                                                                                                                    { gr::tag::TRIGGER_TIME, static_cast<uint64_t>(now.count()) } });
+            triggerTags.emplace_back(static_cast<int64_t>(ps_state.produced_worker + triggerOffset), gr::property_map{ // TODO use data from timing message
+                                                                                                                       { gr::tag::TRIGGER_NAME, "PPS" },
+                                                                                                                       { gr::tag::TRIGGER_TIME, static_cast<uint64_t>(now.count()) } });
         }
 
-        for (auto &channel : state.channels) {
+        for (auto &channel : ps_state.channels) {
             const auto writeSignalInfo = !channel.signal_info_written;
             const auto tagsToWrite     = triggerTags.size() + (writeSignalInfo ? 1 : 0);
             if (tagsToWrite == 0) {
                 continue;
             }
             auto writeTags = channel.tag_writer.reserve_output_range(tagsToWrite);
             if (writeSignalInfo) {
-                writeTags[0].index            = static_cast<int64_t>(state.produced_worker);
+                writeTags[0].index            = static_cast<int64_t>(ps_state.produced_worker - 1);
```

Comment:
Since the latest changes, this only works with the "- 1", leading to an index of -1 instead of 0 when sent with the first data chunk. Is that correct??

Reply:
`-1` is the correct index for the very first tag. Normally you wouldn't need to (and shouldn't) publish this tag explicitly but it's implicitly generated via the property map that is given to the make block routine of the Graph and/or the Block<T> constructor or init(..) functions that take the settings as `const property_map &`

Reply:
We have per-channel settings, and need to do it manually because of the custom work() function, so I think it's needed here. One could argue whether the index should be always -1 (wildcard).